Assets/MRTK/Providers/WindowsMixedReality/Shared/InputHandlers/HPMotionControllerInputHandler.cs

```diff
@@ -0,0 +1,263 @@
+                            break;
+                        case DeviceInputType.Menu:
+                            button = ControllerInput.Menu;
+                            break;
+                        case DeviceInputType.ThumbStickTouch:
+                        case DeviceInputType.ThumbStickPress:
+                            button = ControllerInput.Thumbstick;
+                            break;
+                        default:
+                            return;
+                    }
+
+
+                    var buttonData = controllerState.CurrentReading.GetPressedValue(button);
+                    interactionMapping.BoolData = buttonData > 0.0f;
+                }
+
+                // If our value changed raise it.
+                if (interactionMapping.Changed)
+                {
+                    // Raise input system event if it's enabled
+                    if (interactionMapping.BoolData)
+                    {
+                        CoreServices.InputSystem?.RaiseOnInputDown(InputSource, ControllerHandedness, interactionMapping.MixedRealityInputAction);
+                    }
+                    else
+                    {
+                        CoreServices.InputSystem?.RaiseOnInputUp(InputSource, ControllerHandedness, interactionMapping.MixedRealityInputAction);
+                    }
+                }
+            }
+        }
+
+        private static readonly ProfilerMarker UpdateSingleAxisDataPerfMarker = new ProfilerMarker("[MRTK] HPMotionControllerInputHander.UpdateSingleAxisData");
+
+        /// <summary>
+        /// Update an interaction float data type from a SingleAxis (float) input
+        /// </summary>
+        /// <remarks>
+        /// Raises a FloatInputChanged event when the float data changes
+        /// </remarks>
+        internal virtual void UpdateSingleAxisData(MixedRealityInteractionMapping interactionMapping, MotionControllerState controllerState)
+        {
+            using (UpdateSingleAxisDataPerfMarker.Auto())
+            {
+                Debug.Assert(interactionMapping.AxisType == AxisType.SingleAxis);
+                // Update the interaction data source
+                switch (interactionMapping.InputType)
+                {
+                    case DeviceInputType.Trigger:
+                        var triggerData = controllerState.CurrentReading.GetPressedValue(ControllerInput.Trigger);
+                        interactionMapping.BoolData = !Mathf.Approximately(triggerData, 0.0f);
+                        break;
+                    case DeviceInputType.Grip:
+                        var gripData = controllerState.CurrentReading.GetPressedValue(ControllerInput.Grasp);
+                        interactionMapping.BoolData = !Mathf.Approximately(gripData, 0.0f);
+                        break;
+                    default:
+                        return;
```

Comment:
If this returns by default, it'll never get to the `switch` on line 209.
Though, those two cases are the same as these. Should one of these `switch`es be removed?